pkg/reconciler/buildrun/resources/sources.go

```diff
@@ -16,15 +16,24 @@ func AmendTaskSpecWithSources(
 	cfg *config.Config,
 	taskSpec *v1beta1.TaskSpec,
 	build *buildv1alpha1.Build,
-) {
+) error {
 	// create the step for spec.source, this is always Git
-	sources.AppendGitStep(cfg, taskSpec, build.Spec.Source, "default")
+	sources.AppendGitSourceStep(cfg, taskSpec, build.Spec.Source, "default")
 
 	// create the step for spec.sources, this will eventually change into different steps depending on the type of the source
 	if build.Spec.Sources != nil {
 		for _, source := range *build.Spec.Sources {
-			// today, we only have HTTP sources
-			sources.AppendHTTPStep(cfg, taskSpec, source)
+			switch source.Type {
+			case buildv1alpha1.BuildSourceTypeGit:
+				if err := sources.AppendGitStep(cfg, taskSpec, source); err != nil {
```

Comment:
So far, with Git only available as `spec.source`, it was always first and going into the empty /workspace/source directory.

Assuming we do not forbid to clone a Git source into an existing directory that is not empty, then we will need to modify the git step implementation as this runs `git clone` directly into the destination which will fail if this is a non-empty directory.